### ISSUE TITLE
Connecting to a discovered sink times out

### ISSUE THREAD

**Issue by dbrandonjohnson** [benzea/gnome-screencast#129]
Fedora 32
RPM Package: gnome-network-displays-0.90.3-1.fc32.x86_64

Network Display is able to discover but the connection times out.

```
Jun 01 14:18:21 carbon.home.open-tec.net systemd[3684]: Started Application launched by gnome-shell.
Jun 01 14:18:22 carbon.home.open-tec.net xdg-desktop-por[1955391]: Unhandled parent window type 
Jun 01 14:18:22 carbon.home.open-tec.net xdg-desktop-por[1955391]: Failed to associate portal window with parent window 
Jun 01 14:18:24 carbon.home.open-tec.net pipewire[229893]: [E][000131266.774128][module-access.c:91 check_flatpak()] failed to open "/proc/1953147/root": Permission denied
Jun 01 14:18:24 carbon.home.open-tec.net pipewire[229893]: [W][000131266.774204][module-access.c:165 context_check_access()] module 0x558b78c68510: client 0x558b78fd2d80 sandbox check failed: Permission denied
Jun 01 14:18:28 carbon.home.open-tec.net wpa_supplicant[2684]: P2P-DEVICE-FOUND 5e:f3:70:1b:72:dd p2p_dev_addr=5e:f3:70:1b:72:dd pri_dev_type=8-0050F204-5 name='HP_Pro_Tablet_Dock 2DD' config_methods=0x88 dev_capab=0x25 group_capab=0xa wfd_dev_info=0x0151022a012c new=1
Jun 01 14:18:28 carbon.home.open-tec.net wpa_supplicant[2684]: dbus: Register peer object '/fi/w1/wpa_supplicant1/Interfaces/8/Peers/5ef3701b72dd'
Jun 01 14:18:34 carbon.home.open-tec.net wpa_supplicant[2684]: P2P-FIND-STOPPED
Jun 01 14:18:34 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042714.5409] device (p2p-dev-wlp2s0): Activation: starting connection 'Wi-Fi P2P Peer 5E:F3:70:1B:72:DD' (27fd6887-ec93-4b6b-92d5-97de31489a9c)
Jun 01 14:18:34 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042714.5411] audit: op="connection-add-activate" uuid="27fd6887-ec93-4b6b-92d5-97de31489a9c" name="Wi-Fi P2P Peer 5E:F3:70:1B:72:DD" pid=1976884 uid=1000 result="success"
Jun 01 14:18:34 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042714.5476] device (p2p-dev-wlp2s0): state change: disconnected -> prepare (reason 'none', sys-iface-state: 'managed')
Jun 01 14:18:34 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042714.5483] device (p2p-dev-wlp2s0): state change: prepare -> config (reason 'none', sys-iface-state: 'managed')
Jun 01 14:18:34 carbon.home.open-tec.net systemd-udevd[1977034]: Using default interface naming scheme 'v245'.
Jun 01 14:18:34 carbon.home.open-tec.net systemd-udevd[1977034]: ethtool: autonegotiation is unset or enabled, the speed and duplex are not writable.
Jun 01 14:18:34 carbon.home.open-tec.net systemd-udevd[1977034]: Could not set AlternativeName= or apply AlternativeNamesPolicy= on p2p-wlp2s0-7: File exists
Jun 01 14:18:34 carbon.home.open-tec.net systemd-udevd[1977034]: p2p-wlp2s0-7: Could not apply link config, ignoring: File exists
Jun 01 14:18:37 carbon.home.open-tec.net ModemManager[1895]: <info>  Couldn't check support for device '/sys/devices/pci0000:00/0000:00:1c.0/0000:02:00.0': not supported by any plugin
Jun 01 14:18:37 carbon.home.open-tec.net wpa_supplicant[2684]: P2P-GO-NEG-SUCCESS role=GO freq=5745 ht40=0 peer_dev=5e:f3:70:1b:72:dd peer_iface=5e:f3:70:1b:72:dd wps_method=PBC
Jun 01 14:18:37 carbon.home.open-tec.net wpa_supplicant[2684]: Using interface p2p-wlp2s0-7 with hwaddr 3c:6a:a7:b0:6a:45 and ssid "DIRECT-5H"
Jun 01 14:18:37 carbon.home.open-tec.net kernel: IPv6: ADDRCONF(NETDEV_CHANGE): p2p-wlp2s0-7: link becomes ready
Jun 01 14:18:37 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: interface state UNINITIALIZED->ENABLED
Jun 01 14:18:37 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: AP-ENABLED
Jun 01 14:18:37 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: CTRL-EVENT-CONNECTED - Connection to 3c:6a:a7:b0:6a:45 completed [id=0 id_str=]
Jun 01 14:18:37 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: WPS-PBC-ACTIVE
Jun 01 14:18:39 carbon.home.open-tec.net avahi-daemon[1897]: Joining mDNS multicast group on interface p2p-wlp2s0-7.IPv6 with address fe80::3e6a:a7ff:feb0:6a45.
Jun 01 14:18:39 carbon.home.open-tec.net avahi-daemon[1897]: New relevant interface p2p-wlp2s0-7.IPv6 for mDNS.
Jun 01 14:18:39 carbon.home.open-tec.net avahi-daemon[1897]: Registering new address record for fe80::3e6a:a7ff:feb0:6a45 on p2p-wlp2s0-7.*.
Jun 01 14:18:39 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: CTRL-EVENT-EAP-STARTED 5e:f3:70:1b:72:dd
Jun 01 14:18:39 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: CTRL-EVENT-EAP-PROPOSED-METHOD vendor=0 method=1
Jun 01 14:18:39 carbon.home.open-tec.net wpa_supplicant[2684]: wlp2s0: CTRL-EVENT-SIGNAL-CHANGE above=1 signal=-32 noise=9999 txrate=400000
Jun 01 14:18:39 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: CTRL-EVENT-EAP-STARTED 5e:f3:70:1b:72:dd
Jun 01 14:18:39 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: CTRL-EVENT-EAP-PROPOSED-METHOD vendor=0 method=1
Jun 01 14:18:39 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: CTRL-EVENT-EAP-PROPOSED-METHOD vendor=14122 method=254
Jun 01 14:18:41 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: WPS-REG-SUCCESS 5e:f3:70:1b:72:dd 5ef3701b-72dd-0000-0000-000000000000
Jun 01 14:18:41 carbon.home.open-tec.net wpa_supplicant[2684]: P2P-GROUP-FORMATION-SUCCESS
Jun 01 14:18:41 carbon.home.open-tec.net wpa_supplicant[2684]: P2P-GROUP-STARTED p2p-wlp2s0-7 GO ssid="DIRECT-5H" freq=5745 go_dev_addr=3c:6a:a7:b0:6a:44
Jun 01 14:18:41 carbon.home.open-tec.net wpa_supplicant[2684]: dbus: Register group object '/fi/w1/wpa_supplicant1/Interfaces/15/Groups/5H'
Jun 01 14:18:41 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: WPS-PBC-DISABLE
Jun 01 14:18:41 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: WPS-SUCCESS
Jun 01 14:18:41 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: CTRL-EVENT-EAP-FAILURE 5e:f3:70:1b:72:dd
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.8703] sup-iface[0x558d3ab31bf0,(null)]: supports 5 scan SSIDs
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.8822] device (p2p-dev-wlp2s0): P2P Group supplicant interface state: starting -> ready
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9179] device (p2p-dev-wlp2s0): state change: config -> ip-config (reason 'none', sys-iface-state: 'managed')
Jun 01 14:18:41 carbon.home.open-tec.net avahi-daemon[1897]: Withdrawing address record for fe80::3e6a:a7ff:feb0:6a45 on p2p-wlp2s0-7.
Jun 01 14:18:41 carbon.home.open-tec.net avahi-daemon[1897]: Leaving mDNS multicast group on interface p2p-wlp2s0-7.IPv6 with address fe80::3e6a:a7ff:feb0:6a45.
Jun 01 14:18:41 carbon.home.open-tec.net avahi-daemon[1897]: Interface p2p-wlp2s0-7.IPv6 no longer relevant for mDNS.
Jun 01 14:18:41 carbon.home.open-tec.net avahi-daemon[1897]: Joining mDNS multicast group on interface p2p-wlp2s0-7.IPv6 with address fe80::80db:c605:d4cd:bb59.
Jun 01 14:18:41 carbon.home.open-tec.net avahi-daemon[1897]: New relevant interface p2p-wlp2s0-7.IPv6 for mDNS.
Jun 01 14:18:41 carbon.home.open-tec.net avahi-daemon[1897]: Registering new address record for fe80::80db:c605:d4cd:bb59 on p2p-wlp2s0-7.*.
Jun 01 14:18:41 carbon.home.open-tec.net avahi-daemon[1897]: Joining mDNS multicast group on interface p2p-wlp2s0-7.IPv4 with address 10.42.0.1.
Jun 01 14:18:41 carbon.home.open-tec.net avahi-daemon[1897]: New relevant interface p2p-wlp2s0-7.IPv4 for mDNS.
Jun 01 14:18:41 carbon.home.open-tec.net avahi-daemon[1897]: Registering new address record for 10.42.0.1 on p2p-wlp2s0-7.IPv4.
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9398] Executing: /usr/sbin/iptables --table filter --insert INPUT --in-interface p2p-wlp2s0-7 --protocol tcp --destination-port 53 --jump ACCEPT
Jun 01 14:18:41 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=4
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9418] Executing: /usr/sbin/iptables --table filter --insert INPUT --in-interface p2p-wlp2s0-7 --protocol udp --destination-port 53 --jump ACCEPT
Jun 01 14:18:41 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=5
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9437] Executing: /usr/sbin/iptables --table filter --insert INPUT --in-interface p2p-wlp2s0-7 --protocol tcp --destination-port 67 --jump ACCEPT
Jun 01 14:18:41 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=6
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9460] Executing: /usr/sbin/iptables --table filter --insert INPUT --in-interface p2p-wlp2s0-7 --protocol udp --destination-port 67 --jump ACCEPT
Jun 01 14:18:41 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=7
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9479] Executing: /usr/sbin/iptables --table filter --insert FORWARD --in-interface p2p-wlp2s0-7 --jump REJECT
Jun 01 14:18:41 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=8
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9497] Executing: /usr/sbin/iptables --table filter --insert FORWARD --out-interface p2p-wlp2s0-7 --jump REJECT
Jun 01 14:18:41 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=9
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9514] Executing: /usr/sbin/iptables --table filter --insert FORWARD --in-interface p2p-wlp2s0-7 --out-interface p2p-wlp2s0-7 --jump ACCEPT
Jun 01 14:18:41 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=10
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9530] Executing: /usr/sbin/iptables --table filter --insert FORWARD --source 10.42.0.0/255.255.255.0 --in-interface p2p-wlp2s0-7 --jump ACCEPT
Jun 01 14:18:41 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=11
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9546] Executing: /usr/sbin/iptables --table filter --insert FORWARD --destination 10.42.0.0/255.255.255.0 --out-interface p2p-wlp2s0-7 --match state --state ESTABLISHED,RELATED --jump ACCEPT
Jun 01 14:18:41 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=12
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9564] Executing: /usr/sbin/iptables --table nat --insert POSTROUTING --source 10.42.0.0/255.255.255.0 ! --destination 10.42.0.0/255.255.255.0 --jump MASQUERADE
Jun 01 14:18:41 carbon.home.open-tec.net audit: NETFILTER_CFG table=nat family=2 entries=5
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9581] dnsmasq-manager: starting dnsmasq...
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9593] device (p2p-dev-wlp2s0): state change: ip-config -> ip-check (reason 'none', sys-iface-state: 'managed')
Jun 01 14:18:41 carbon.home.open-tec.net dnsmasq[1977066]: listening on p2p-wlp2s0-7(#17): 10.42.0.1 port 53
Jun 01 14:18:41 carbon.home.open-tec.net dnsmasq[1977066]: started, version 2.81 cachesize 150
Jun 01 14:18:41 carbon.home.open-tec.net dnsmasq[1977066]: compile time options: IPv6 GNU-getopt DBus no-UBus no-i18n IDN2 DHCP DHCPv6 no-Lua TFTP no-conntrack ipset auth DNSSEC loop-detect inotify dumpfile
Jun 01 14:18:41 carbon.home.open-tec.net dnsmasq[1977066]: chown of PID file /run/nm-dnsmasq-p2p-wlp2s0-7.pid failed: Operation not permitted
Jun 01 14:18:41 carbon.home.open-tec.net dnsmasq-dhcp[1977066]: DHCP, IP range 10.42.0.10 -- 10.42.0.254, lease time 1h
Jun 01 14:18:41 carbon.home.open-tec.net dnsmasq[1977066]: reading /etc/resolv.conf
Jun 01 14:18:41 carbon.home.open-tec.net dnsmasq[1977066]: using nameserver 127.0.0.1#53
Jun 01 14:18:41 carbon.home.open-tec.net dnsmasq[1977066]: cleared cache
Jun 01 14:18:41 carbon.home.open-tec.net systemd[1]: Starting Network Manager Script Dispatcher Service...
Jun 01 14:18:41 carbon.home.open-tec.net systemd[1]: Started Network Manager Script Dispatcher Service.
Jun 01 14:18:41 carbon.home.open-tec.net audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=NetworkManager-dispatcher comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9759] device (p2p-dev-wlp2s0): state change: ip-check -> secondaries (reason 'none', sys-iface-state: 'managed')
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9762] device (p2p-dev-wlp2s0): state change: secondaries -> activated (reason 'none', sys-iface-state: 'managed')
Jun 01 14:18:41 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042721.9776] device (p2p-dev-wlp2s0): Activation: successful, device activated.
Jun 01 14:18:41 carbon.home.open-tec.net systemd[1]: iscsi.service: Unit cannot be reloaded because it is inactive.
Jun 01 14:18:42 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: AP-STA-CONNECTED 5e:f3:70:1b:72:dd p2p_dev_addr=5e:f3:70:1b:72:dd
Jun 01 14:18:42 carbon.home.open-tec.net wpa_supplicant[2684]: AP-STA-CONNECTED 5e:f3:70:1b:72:dd p2p_dev_addr=5e:f3:70:1b:72:dd
Jun 01 14:18:43 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591042723.8270] policy: ipv6-pd: no device to obtain a subnet to share on p2p-dev-wlp2s0 from
Jun 01 14:18:47 carbon.home.open-tec.net dnsmasq-dhcp[1977066]: DHCPDISCOVER(p2p-wlp2s0-7) 5e:f3:70:1b:72:dd
Jun 01 14:18:47 carbon.home.open-tec.net dnsmasq-dhcp[1977066]: DHCPOFFER(p2p-wlp2s0-7) 10.42.0.63 5e:f3:70:1b:72:dd
Jun 01 14:18:47 carbon.home.open-tec.net dnsmasq-dhcp[1977066]: DHCPDISCOVER(p2p-wlp2s0-7) 5e:f3:70:1b:72:dd
Jun 01 14:18:47 carbon.home.open-tec.net dnsmasq-dhcp[1977066]: DHCPOFFER(p2p-wlp2s0-7) 10.42.0.63 5e:f3:70:1b:72:dd
Jun 01 14:18:47 carbon.home.open-tec.net dnsmasq-dhcp[1977066]: DHCPDISCOVER(p2p-wlp2s0-7) 5e:f3:70:1b:72:dd
Jun 01 14:18:47 carbon.home.open-tec.net dnsmasq-dhcp[1977066]: DHCPOFFER(p2p-wlp2s0-7) 10.42.0.63 5e:f3:70:1b:72:dd
Jun 01 14:18:47 carbon.home.open-tec.net dnsmasq-dhcp[1977066]: DHCPDISCOVER(p2p-wlp2s0-7) 5e:f3:70:1b:72:dd
Jun 01 14:18:47 carbon.home.open-tec.net dnsmasq-dhcp[1977066]: DHCPOFFER(p2p-wlp2s0-7) 10.42.0.63 5e:f3:70:1b:72:dd
Jun 01 14:18:47 carbon.home.open-tec.net dnsmasq-dhcp[1977066]: DHCPREQUEST(p2p-wlp2s0-7) 10.42.0.63 5e:f3:70:1b:72:dd
Jun 01 14:18:47 carbon.home.open-tec.net dnsmasq-dhcp[1977066]: DHCPACK(p2p-wlp2s0-7) 10.42.0.63 5e:f3:70:1b:72:dd
Jun 01 14:18:52 carbon.home.open-tec.net systemd[1]: NetworkManager-dispatcher.service: Succeeded.
Jun 01 14:18:52 carbon.home.open-tec.net audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=NetworkManager-dispatcher comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
---REDACTED NOT RELEVENT LOGS
Jun 01 14:25:18 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: AP-STA-DISCONNECTED 5e:f3:70:1b:72:dd p2p_dev_addr=5e:f3:70:1b:72:dd
Jun 01 14:25:18 carbon.home.open-tec.net wpa_supplicant[2684]: AP-STA-DISCONNECTED 5e:f3:70:1b:72:dd p2p_dev_addr=5e:f3:70:1b:72:dd
Jun 01 14:25:38 carbon.home.open-tec.net wpa_supplicant[2684]: P2P-DEVICE-LOST p2p_dev_addr=5e:f3:70:1b:72:dd
Jun 01 14:25:38 carbon.home.open-tec.net wpa_supplicant[2684]: dbus: Unregister peer object '/fi/w1/wpa_supplicant1/Interfaces/8/Peers/5ef3701b72dd'
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <warn>  [1591043143.7339] device (p2p-dev-wlp2s0): Peer requested in connection is missing for too long, failing connection.
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591043143.7341] device (p2p-dev-wlp2s0): state change: activated -> failed (reason 'peer-not-found', sys-iface-state: 'managed')
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <warn>  [1591043143.7421] device (p2p-dev-wlp2s0): Activation: failed for connection 'Wi-Fi P2P Peer 5E:F3:70:1B:72:DD'
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591043143.7489] device (p2p-dev-wlp2s0): state change: failed -> disconnected (reason 'none', sys-iface-state: 'managed')
Jun 01 14:25:43 carbon.home.open-tec.net avahi-daemon[1897]: Withdrawing address record for fe80::80db:c605:d4cd:bb59 on p2p-wlp2s0-7.
Jun 01 14:25:43 carbon.home.open-tec.net avahi-daemon[1897]: Leaving mDNS multicast group on interface p2p-wlp2s0-7.IPv6 with address fe80::80db:c605:d4cd:bb59.
Jun 01 14:25:43 carbon.home.open-tec.net avahi-daemon[1897]: Interface p2p-wlp2s0-7.IPv6 no longer relevant for mDNS.
Jun 01 14:25:43 carbon.home.open-tec.net avahi-daemon[1897]: Withdrawing address record for 10.42.0.1 on p2p-wlp2s0-7.
Jun 01 14:25:43 carbon.home.open-tec.net avahi-daemon[1897]: Leaving mDNS multicast group on interface p2p-wlp2s0-7.IPv4 with address 10.42.0.1.
Jun 01 14:25:43 carbon.home.open-tec.net dnsmasq[1977066]: exiting on receipt of SIGTERM
Jun 01 14:25:43 carbon.home.open-tec.net wpa_supplicant[2684]: P2P-GROUP-REMOVED p2p-wlp2s0-7 GO reason=REQUESTED
Jun 01 14:25:43 carbon.home.open-tec.net avahi-daemon[1897]: Interface p2p-wlp2s0-7.IPv4 no longer relevant for mDNS.
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591043143.7705] Executing: /usr/sbin/iptables --table nat --delete POSTROUTING --source 10.42.0.0/255.255.255.0 ! --destination 10.42.0.0/255.255.255.0 --jump MASQUERADE
Jun 01 14:25:43 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: interface state ENABLED->DISABLED
Jun 01 14:25:43 carbon.home.open-tec.net systemd[1]: Starting Network Manager Script Dispatcher Service...
Jun 01 14:25:43 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: AP-DISABLED
Jun 01 14:25:43 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: CTRL-EVENT-DISCONNECTED bssid=3c:6a:a7:b0:6a:45 reason=3 locally_generated=1
Jun 01 14:25:43 carbon.home.open-tec.net audit: NETFILTER_CFG table=nat family=2 entries=6
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591043143.7834] Executing: /usr/sbin/iptables --table filter --delete FORWARD --destination 10.42.0.0/255.255.255.0 --out-interface p2p-wlp2s0-7 --match state --state ESTABLISHED,RELATED --jump ACCEPT
Jun 01 14:25:43 carbon.home.open-tec.net wpa_supplicant[2684]: nl80211: deinit ifname=p2p-wlp2s0-7 disabled_11b_rates=1
Jun 01 14:25:43 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=13
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591043143.7966] Executing: /usr/sbin/iptables --table filter --delete FORWARD --source 10.42.0.0/255.255.255.0 --in-interface p2p-wlp2s0-7 --jump ACCEPT
Jun 01 14:25:43 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=12
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591043143.8067] Executing: /usr/sbin/iptables --table filter --delete FORWARD --in-interface p2p-wlp2s0-7 --out-interface p2p-wlp2s0-7 --jump ACCEPT
Jun 01 14:25:43 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=11
Jun 01 14:25:43 carbon.home.open-tec.net systemd[1]: Started Network Manager Script Dispatcher Service.
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591043143.8149] Executing: /usr/sbin/iptables --table filter --delete FORWARD --out-interface p2p-wlp2s0-7 --jump REJECT
Jun 01 14:25:43 carbon.home.open-tec.net audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=NetworkManager-dispatcher comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Jun 01 14:25:43 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=10
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591043143.8207] Executing: /usr/sbin/iptables --table filter --delete FORWARD --in-interface p2p-wlp2s0-7 --jump REJECT
Jun 01 14:25:43 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=9
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591043143.8249] Executing: /usr/sbin/iptables --table filter --delete INPUT --in-interface p2p-wlp2s0-7 --protocol udp --destination-port 67 --jump ACCEPT
Jun 01 14:25:43 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=8
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591043143.8346] Executing: /usr/sbin/iptables --table filter --delete INPUT --in-interface p2p-wlp2s0-7 --protocol tcp --destination-port 67 --jump ACCEPT
Jun 01 14:25:43 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=7
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591043143.8456] Executing: /usr/sbin/iptables --table filter --delete INPUT --in-interface p2p-wlp2s0-7 --protocol udp --destination-port 53 --jump ACCEPT
Jun 01 14:25:43 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=6
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591043143.8506] Executing: /usr/sbin/iptables --table filter --delete INPUT --in-interface p2p-wlp2s0-7 --protocol tcp --destination-port 53 --jump ACCEPT
Jun 01 14:25:43 carbon.home.open-tec.net audit: NETFILTER_CFG table=filter family=2 entries=5
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <warn>  [1591043143.8569] sup-iface: failed to cancel p2p connect: P2P cancel failed
Jun 01 14:25:43 carbon.home.open-tec.net NetworkManager[2071]: <info>  [1591043143.8571] device (p2p-dev-wlp2s0): P2P Group supplicant interface state: ready -> down
Jun 01 14:25:53 carbon.home.open-tec.net systemd[1]: NetworkManager-dispatcher.service: Succeeded.
Jun 01 14:25:53 carbon.home.open-tec.net audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=NetworkManager-dispatcher comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
```

Let me know if there are additional logs or debugging steps that would help track down this issue.

**Comment by benzea** [comment]
Have you checked the hints in README?

**Comment by benzea** [comment]
Anyway:

```
Jun 01 14:18:47 carbon.home.open-tec.net dnsmasq-dhcp[1977066]: DHCPACK(p2p-wlp2s0-7) 10.42.0.63 5e:f3:70:1b:72:dd
Jun 01 14:18:52 carbon.home.open-tec.net systemd[1]: NetworkManager-dispatcher.service: Succeeded.
Jun 01 14:25:18 carbon.home.open-tec.net wpa_supplicant[2684]: p2p-wlp2s0-7: AP-STA-DISCONNECTED 5e:f3:70:1b:72:dd p2p_dev_addr=5e:f3:70:1b:72:dd
```

The TV/whatever simply disconnects again. The question is why …

**Comment by dbrandonjohnson** [comment]
G_MESSAGES_DEBUG=all gnome-network-displays output: 
```
(gnome-network-displays:2029432): GLib-GIO-DEBUG: 09:42:54.257: _g_io_module_get_default: Found default implementation dconf (DConfSettingsBackend) for ‘gsettings-backend’
(gnome-network-displays:2029432): dconf-DEBUG: 09:42:54.257: watch_fast: "/org/gnome/desktop/interface/" (establishing: 0, active: 0)
(gnome-network-displays:2029432): dconf-DEBUG: 09:42:54.258: watch_fast: "/org/gnome/settings-daemon/peripherals/mouse/" (establishing: 0, active: 0)
(gnome-network-displays:2029432): dconf-DEBUG: 09:42:54.258: watch_fast: "/org/gnome/desktop/sound/" (establishing: 0, active: 0)
(gnome-network-displays:2029432): dconf-DEBUG: 09:42:54.258: watch_fast: "/org/gnome/desktop/privacy/" (establishing: 0, active: 0)
(gnome-network-displays:2029432): dconf-DEBUG: 09:42:54.258: watch_fast: "/org/gnome/desktop/wm/preferences/" (establishing: 0, active: 0)
(gnome-network-displays:2029432): dconf-DEBUG: 09:42:54.258: watch_fast: "/org/gnome/settings-daemon/plugins/xsettings/" (establishing: 0, active: 0)
(gnome-network-displays:2029432): dconf-DEBUG: 09:42:54.258: watch_fast: "/org/gnome/desktop/a11y/" (establishing: 0, active: 0)
(gnome-network-displays:2029432): dconf-DEBUG: 09:42:54.260: watch_established: "/org/gnome/desktop/interface/" (establishing: 1)
(gnome-network-displays:2029432): dconf-DEBUG: 09:42:54.260: watch_established: "/org/gnome/settings-daemon/peripherals/mouse/" (establishing: 1)
(gnome-network-displays:2029432): dconf-DEBUG: 09:42:54.260: watch_established: "/org/gnome/desktop/sound/" (establishing: 1)
(gnome-network-displays:2029432): dconf-DEBUG: 09:42:54.260: watch_established: "/org/gnome/desktop/privacy/" (establishing: 1)
(gnome-network-displays:2029432): dconf-DEBUG: 09:42:54.260: watch_established: "/org/gnome/desktop/wm/preferences/" (establishing: 1)
(gnome-network-displays:2029432): dconf-DEBUG: 09:42:54.261: watch_established: "/org/gnome/settings-daemon/plugins/xsettings/" (establishing: 1)
(gnome-network-displays:2029432): dconf-DEBUG: 09:42:54.261: watch_established: "/org/gnome/desktop/a11y/" (establishing: 1)
(gnome-network-displays:2029432): GLib-GIO-DEBUG: 09:42:54.293: _g_io_module_get_default: Found default implementation gvfs (GDaemonVfs) for ‘gio-vfs’
(gnome-network-displays:2029432): GLib-DEBUG: 09:42:54.317: unsetenv() is not thread-safe and should not be used after threads are created
(gnome-network-displays:2029432): Gtk-DEBUG: 09:42:54.317: Connecting to session manager
** (gnome-network-displays:2029432): DEBUG: 09:42:54.388: NdScreencastPortal: Aquired Portal proxy
** (gnome-network-displays:2029432): DEBUG: 09:42:54.389: task is: 0x7fbf4800eca0
** (gnome-network-displays:2029432): DEBUG: 09:42:54.393: NdPulseaudio: Querying sink info by name
** (gnome-network-displays:2029432): DEBUG: 09:42:54.393: NdPulseaudio: Error querying sink info
** (gnome-network-displays:2029432): DEBUG: 09:42:54.393: NdPulseaudio: Got a sink info for the expected name
** (gnome-network-displays:2029432): DEBUG: 09:42:54.399: WFDP2PRegistry: Found a new device, creating provider
** (gnome-network-displays:2029432): DEBUG: 09:42:54.399: WfdP2PProvider: Discover is now set to 1
** (gnome-network-displays:2029432): DEBUG: 09:42:54.422: WFDP2PRegistry: Got NMClient
** (gnome-network-displays:2029432): DEBUG: 09:42:54.428: NdScreencastPortal: Received CreateSession response
** (gnome-network-displays:2029432): DEBUG: 09:42:54.428: simple variant lookup: /org/freedesktop/portal/desktop/session/1_875/gtk930774899
** (gnome-network-displays:2029432): DEBUG: 09:42:54.456: NdScreencastPortal: Received SelectSource response
** (gnome-network-displays:2029432): DEBUG: 09:42:58.157: WFDP2PProvider: Found a new sink with peer 0x55aa50931b00 on device 0x55aa5090a2a0
** (gnome-network-displays:2029432): DEBUG: 09:42:58.158: SinkList: Adding a sink
** (gnome-network-displays:2029432): DEBUG: 09:43:01.030: NdScreencastPortal: Received Start response
** (gnome-network-displays:2029432): DEBUG: 09:43:01.030: Got a stream with node ID: 100
** (gnome-network-displays:2029432): DEBUG: 09:43:02.106: Found openh264enc for video encoding.
** (gnome-network-displays:2029432): DEBUG: 09:43:02.106: Found x264enc for video encoding.
** (gnome-network-displays:2029432): DEBUG: 09:43:02.106: Found fdkaacenc for audio encoding.
** (gnome-network-displays:2029432): DEBUG: 09:43:02.106: Found avenc_aac for audio encoding.
** (gnome-network-displays:2029432): DEBUG: 09:43:02.107: Got state change notification from streaming sink to state ND_SINK_STATE_WAIT_P2P
** (gnome-network-displays:2029432): DEBUG: 09:43:02.109: WfdP2PProvider: Discover is now set to 0
** (gnome-network-displays:2029432): DEBUG: 09:43:02.166: NdWfdP2PSink: Got P2P connection
** (gnome-network-displays:2029432): DEBUG: 09:43:02.166: Found openh264enc for video encoding.
** (gnome-network-displays:2029432): DEBUG: 09:43:02.166: Found x264enc for video encoding.
** (gnome-network-displays:2029432): DEBUG: 09:43:02.166: Found fdkaacenc for audio encoding.
** (gnome-network-displays:2029432): DEBUG: 09:43:02.166: Found avenc_aac for audio encoding.
** (gnome-network-displays:2029432): DEBUG: 09:43:02.167: Got state change notification from streaming sink to state ND_SINK_STATE_WAIT_SOCKET
** (gnome-network-displays:2029432): DEBUG: 09:50:09.658: WFDP2PProvider: Peer removed
** (gnome-network-displays:2029432): DEBUG: 09:50:09.658: MetaSink: No usable sink is left, object has become invalid.
** (gnome-network-displays:2029432): DEBUG: 09:50:09.658: SinkList: Removing a sink
** (gnome-network-displays:2029432): DEBUG: 09:50:14.745: WfdServer: Finalize
** (gnome-network-displays:2029432): DEBUG: 09:50:14.745: WfdMediaFactory: Finalize
** (gnome-network-displays:2029432): DEBUG: 09:50:14.745: Got state change notification from streaming sink to state ND_SINK_STATE_ERROR
```


I don't know for sure but I think the Miracast device I'm using (it is an HP wireless dock) it might be waiting on a pin.

**Comment by benzea** [comment]
Right, we only implement the PBC method currently (in NetworkManager). i.e. where you have to push a button (or the device just assumes it is pushed and lets you connect). If this device requires the use of the PIN method, then it would not be usable with gnome-network-displays currently due to the restrictions in NetworkManager.

Technically this wouldn't be a gnome-network-displays bug. When implemented in NetworkManager, you would likely simply get the existing GNOME Shell integration to prompt for the PIN.
